---
title: "Lab_10"
author: "Abigail Booth"
date: "November 2, 2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Part I: Reading VCF data

```{r, message=FALSE}
library(vcfR)
```
```{r}
data(vcfR_example)
vcf
```

### The meta region

```{r}
strwrap(vcf@meta[1:7])
```
```{r}
queryMETA(vcf)
```
```{r}
queryMETA(vcf, element = 'DP')
```
```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```

### The fix region 

```{r}
head(getFIX(vcf))
```

### The gt region

```{r}
vcf@gt[1:6, 1:4]
```

### vcfR

```{r}
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
```
```{r}
head(vcf)
```
```{r}
write.vcf(vcf, "myVCFdata_filtered.vcf.gz")
```

### Exercises part I

1. We would find more information about readvcfR() by typing ?readvcfR() into the console or in a code chunk.

2. We could learn what the acronym "AD" stands for with 
```{r}
queryMETA(vcf, element = 'AD')
```

3. We could view the last few lines of the fix data using tail:
```{r}
tail(vcf@fix)
```

4. QUAL does not seem important because all of the data points are of similar quality so there are no outliers that might otherwise skew the data. 
```{r}
plot(vcf)
```

5. We would query the sample names with:
```{r}
colnames(vcf@gt)
```

# Part II: Analysis of genome data

### Opening and examining the dataset

```{r}
library('vcfR')
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
vcf
```

### Converting VCF data to a genlight object

```{r}
x <- vcfR2genlight(vcf)
x
```
```{r}
# vcfR
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```
```{r}
# genlight
t(as.matrix(x))[c(1,5,17), 1:3]
```
```{r, message=FALSE, warning=FALSE}
library(adegenet)
```
```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```
```{r}
ploidy(x) <- 2
```

### Distance matrices

```{r}
x.dist <- dist(x)
```
```{r, message=FALSE, warning=FALSE}
x.dist <- poppr::bitwise.dist(x)
```

### Creating chromR objects

```{r}
library(vcfR)

# Find the files.
vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
dna_file <- system.file("extdata", "pinf_sc50.fasta", package = "pinfsc50")
gff_file <- system.file("extdata", "pinf_sc50.gff", package = "pinfsc50")

# Input the files.
vcf <- read.vcfR(vcf_file, verbose = FALSE)
dna <- ape::read.dna(dna_file, format = "fasta")
gff <- read.table(gff_file, sep="\t", quote="")

# Create a chromR object.
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=TRUE)
chrom
```
```{r}
plot(chrom)
```
```{r}
chromoqc(chrom, dp.alpha = 66)
```

### Processing chromR objects

```{r}
chrom <- proc.chromR(chrom, verbose = TRUE)
```
```{r}
plot(chrom)
```
```{r}
chromoqc(chrom, dp.alpha = 66)
```
```{r}
#vcf <- read.vcfR("pinfsc50_qc.vcf.gz", verbose = FALSE)
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz", verbose = FALSE)
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom <- proc.chromR(chrom, verbose = FALSE)
chromoqc(chrom, dp.alpha = 66)
```

### Tabular summaries

```{r}
head(chrom@var.info)
```
```{r}
head(chrom@win.info)
```

### Genetic differentiation

```{r}
library(vcfR)
data(vcfR_example)
pop <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
myDiff <- genetic_diff(vcf, pops = pop, method = 'nei')
knitr::kable(head(myDiff[,1:15]))
```
```{r}
knitr::kable(head(myDiff[,16:19]))
```
```{r}
knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))
```
```{r, message=FALSE}
library(reshape2)
library(ggplot2)

dpf <- melt(myDiff[,c(3:8,19)], varnames=c('Index', 'Sample'), value.name = 'Depth', na.rm=TRUE)
```
```{r}
p <- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill="#2ca25f", adjust = 1.2)
p <- p + xlab("")
p <- p + ylab("")
p <- p + theme_bw()
p
```

### Exercises part II

1. 
```{r}
plot(getPOS(vcf), myDiff$Gprimest, xaxt="n", xlab="Genomic Position", ylab=expression(italic("G'"["ST"])), col="purple") 
axis(side = 1, at = seq(0, 1e5, by = 1e4), labels = seq(0, 101, by = 10))
```

2. FST, GST, and G'ST all related to population differentiation and heterozygosity, so the sample size for the Manhattan plot is very low, which makes the plot non-representative as can be seen with:
```{r}
table(pop)
```

3. 
```{r}
chromoqc(chrom, boxp = TRUE, dp.alpha = 50, xlim = c(2e05, 6e05))
```

4. 
```{r}
queryMETA(vcf, nice=TRUE)
```